packages/salesforcedx-vscode-core/src/commands/commands.ts

```diff
@@ -118,6 +120,74 @@ export type DirFileNameSelection = {
   outputdir: string;
 };
 
+export class SelectFileName
+  implements ParametersGatherer<{ fileName: string }> {
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<{ fileName: string }>
+  > {
+    const fileNameInputOptions = <vscode.InputBoxOptions>{
+      prompt: nls.localize('parameter_gatherer_enter_file_name')
+    };
+    const fileName = await vscode.window.showInputBox(fileNameInputOptions);
+    return fileName
+      ? { type: 'CONTINUE', data: { fileName } }
+      : { type: 'CANCEL' };
+  }
+}
+
+export class SelectDirPath
+  implements ParametersGatherer<{ outputdir: string }> {
+  private explorerDir: string | undefined;
+  private globKeyWord: string | undefined;
+
+  public constructor(explorerDir?: { path: string }, globKeyWord?: string) {
+    this.explorerDir = explorerDir ? explorerDir.path : explorerDir;
+    this.globKeyWord = globKeyWord;
+  }
+
+  public globDirs(srcPath: string, priorityKeyword?: string): string[] {
+    const unprioritizedRelDirs = new glob.GlobSync(
+      path.join(srcPath, '**/')
+    ).found.map(value => {
+      let relativePath = path.relative(srcPath, path.join(value, '/'));
+      relativePath = path.join(relativePath, '');
+      return relativePath;
+    });
+    if (priorityKeyword) {
+      const notPrioritized: string[] = [];
+      const prioritized = unprioritizedRelDirs.filter(dir => {
+        if (dir.includes(priorityKeyword)) {
+          return true;
+        } else {
+          notPrioritized.push(dir);
+        }
+      });
+      return prioritized.concat(notPrioritized);
+    }
+    return unprioritizedRelDirs;
+  }
+
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<{ outputdir: string }>
+  > {
+    const rootPath = vscode.workspace.rootPath;
+    let outputdir;
+    if (rootPath) {
+      outputdir = this.explorerDir
+        ? path.relative(rootPath, this.explorerDir)
```

Comment:
After changing to relative path, I forgot to change the explorer dir path to be relative as well. This broke opening the file after right click creation because I create the absolute by appending the outputdir to the workspace dir.